add-pkg.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash -x
+
+BUILDCONF="/usr/share/mix/builder.conf"
+
+# Auto generate builder.conf if it's not there
+if [ ! -f "$BUILDCONF" ]; then
+echo "[Builder]
+SERVER_STATE_DIR = /usr/share/mix/update
+BUNDLE_DIR = /usr/share/mix/mix-bundles
+YUM_CONF = /usr/share/mix/.yum-mix.conf
+CERT = /usr/share/mix/Swupd_Root.pem
+VERSIONS_PATH =/usr/share/mix
+RPMDIR = /usr/share/mix/rpms
+REPODIR = /usr/share/mix/local
+
+[swupd]
+BUNDLE=os-core-update
+CONTENTURL=file:///usr/share/mix/update/www
+VERSIONURL=file:///usr/share/mix/update/www
+FORMAT=1" | sudo tee "$BUILDCONF" > /dev/null
+fi
+
+MIX_DIR="/usr/share/mix/"
+OUTPUTDIR="$MIX_DIR/update/www"
+BUNDLE_DIR="$(awk -F '=' '/BUNDLE_DIR/ {print $2}' $BUILDCONF  | tr -d ' ')"
+RPM_DIR="$(awk -F '=' '/RPMDIR/ {print $2}' $BUILDCONF  | tr -d ' ')"
+REPO_DIR="$(awk -F '=' '/REPODIR/ {print $2}' $BUILDCONF  | tr -d ' ')"
+BUILD_STATE_DIR="/var/lib/mix"
+MIX_CERT="$MIX_DIR/Swupd_Root.pem"
+PRIVKEY="$MIX_DIR/private.pem"
+FORMAT="$(awk -F '=' '/FORMAT/ {print $2}' $BUILDCONF  | tr -d ' ')"
+
+pkg=$1
+bundle=$2
+
+# Make sure all the directories exist
+sudo mkdir -p $MIX_DIR $BUNDLE_DIR $REPO_DIR $BUILD_STATE_DIR
+if [ ! -f "$MIX_DIR/.clearversion" ]; then
+	awk -F  "=" '/^VERSION_ID=/ { print $2 }' /usr/lib/os-release | tr -d '\n' | sudo tee "$MIX_DIR/.clearversion" > /dev/null
+elif [ "$pkg" -eq "regenerate" ]; then
+	echo -n "$(($(cat $MIX_DIR/.clearversion)*1000))" | sudo tee "$MIX_DIR/.mixversion" > /dev/null
+fi
+if [ ! -f "$MIX_DIR/.mixversion" ]; then
+	echo -n "$(($(cat $MIX_DIR/.clearversion)*1000))" | sudo tee "$MIX_DIR/.mixversion" > /dev/null
+fi
+
+clearver=$(cat $MIX_DIR/.clearversion | tr -d ' ')
+echo -n "$clearver" | sudo tee "$MIX_DIR/version" > /dev/null
+
+if [[ -n "$pkg" && -n "$bundle" ]]; then
+	echo "Adding $pkg to $bundle..."
+
+	# Write package to bundle definition
+	echo "$pkg" | sudo tee -a "$BUNDLE_DIR/$bundle" > /dev/null
+fi
+
+# Copy os-core from upstream
+sudo -E curl -O "https://raw.githubusercontent.com/clearlinux/clr-bundles/$clearver/bundles/os-core"
```

Comment:
This curl command could hit a server error, so it could (for example) download a 404 response. You can avoid this issue by passing the `-f` option and handling the special curl error exit status 22.

The same comment applies to all other curl commands in this script.

Reply:
Any thoughts on this, @tmarcu ?

Reply:
Yes I need to add -f :)